pandas/core/groupby/grouper.py

```diff
@@ -496,11 +496,16 @@ def __init__(
             # In extant tests, the new self.grouping_vector matches
             #  `index.get_level_values(ilevel)` whenever
             #  mapper is None and isinstance(index, MultiIndex)
+            # TODO: Can you have two levels with the same name?
+            if isinstance(index, MultiIndex):
+                index_level = index.get_level_values(ilevel)
+            else:
+                index_level = index
             (
                 self.grouping_vector,  # Index
                 self._codes,
                 self._group_index,
-            ) = index._get_grouper_for_level(mapper, level=ilevel, dropna=dropna)
+            ) = index_level._get_grouper_for_level(mapper, dropna=dropna)
```

Comment:
I think this change would make `level` unused in `_get_grouper_for_level`

Reply:
It is unused in `Index._get_grouper_for_level` but still "used" in `MultiIndex._get_grouper_for_level`. I say "used" because `MultiIndex._get_grouper_for_level` is now never called by pandas. I think that means it should be safe to remove entirely from MultiIndex (or maybe just raise a NotImplementedError?). Or is it possible someone downstream is using this? In any case, I would like to do that in a follow up. Okay to open an issue instead?

Reply:
> In any case, I would like to do that in a follow up. Okay to open an issue instead?

Sure, sounds good as a follow up

Reply:
@mroeschke - I'm second guessing my change here. Going to run a set of groupby ASVs to see what impact this has.

Reply:
```
       before           after         ratio
     [ead5c756]       [513c3224]
     <main>           <groupby_cat_unobserved>
+         144±6μs          161±4μs     1.12  groupby.GroupByMethods.time_dtype_as_group('uint', 'quantile', 'direct', 1)
-        38.7±3ms       31.1±0.5ms     0.80  groupby.GroupByCythonAggEaDtypes.time_frame_agg('Int32', 'sum')
```

Reply:
@jbrockmendel - you good with this change? Assuming so, any recommendations on what to do with the no-longer-used `MultiIndex._get_grouper_for_level`?

Reply:
if its no longer used let's get rid of it right?

Reply:
Opened #49452